format-common/src/main/java/co/cask/hydrator/format/input/BlobInputFormatter.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright © 2018 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.hydrator.format.input;
+
+import co.cask.cdap.api.data.format.StructuredRecord;
+import co.cask.cdap.api.data.schema.Schema;
+import com.google.common.io.ByteStreams;
+import org.apache.hadoop.fs.FSDataInputStream;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.io.NullWritable;
+import org.apache.hadoop.mapreduce.InputSplit;
+import org.apache.hadoop.mapreduce.RecordReader;
+import org.apache.hadoop.mapreduce.TaskAttemptContext;
+import org.apache.hadoop.mapreduce.lib.input.FileSplit;
+
+import java.io.IOException;
+import java.util.Collections;
+import java.util.Map;
+
+/**
+ * Reads the entire contents of a File into a single record
+ */
+public class BlobInputFormatter implements FileInputFormatter {
+  private final Schema schema;
+
+  BlobInputFormatter(Schema schema) {
+    this.schema = schema;
+  }
+
+  @Override
+  public Map<String, String> getFormatConfig() {
+    return Collections.emptyMap();
+  }
+
+  @Override
+  public RecordReader<NullWritable, StructuredRecord.Builder> create(FileSplit split, TaskAttemptContext context) {
+    if (split.getLength() > Integer.MAX_VALUE) {
```

Comment:
Is there a relation between the split length and the MAX_VALUE for file size greater than 2GB?

Reply:
we can't create a byte[] using a long, which is why it's restricted by integer size. It's also a terrible idea to have a 2gb record.

Reply:
👍